data/blog/how-open-pr-comments-work.md

```diff
@@ -0,0 +1,124 @@
+---
+title: 'How open PR comments work'
+date: '2024-04-18'
+tags: ['integrations', 'github', 'pr-comments']
+draft: false
+summary: "This is how we manage to comment Sentry issues related to code you're modifying in a pull request within seconds."
+images: []
+layout: PostLayout
+canonicalUrl: open-pr-comments
+authors: ['cathyteng']
+---
+
+At Sentry, we always want to bring value to the developer. One area we can do this is through developer workflows, such as pull requests. If you’re already working in a particular area, it’s useful to be informed about issues related to the code you’re changing so you can be proactive in acknowledging and addressing them.
+
+When somebody opens a Github PR, we parse the functions from the PR diff, search for unresolved, unhandled issues related to those functions in the PR files, and make a nice little comment on that PR.
+
+# How it works
+
+![Mermaid diagram of open PR comments](/images/open-pr-comments/how-it-works.png)
+
+## Webhook
+
+When somebody installs the Sentry Github App, we request permissions so that we can consume webhooks for particular events that we would like to handle. We have a webhook for pull request events that does stuff on the Sentry side when somebody interacts with a pull request on Github. So when somebody opens a pull request, we can kick off the open PR comment workflow.
+
+## Comment workflow
+
+### Qualification checks
+
+We fetch information for the PR files from Github, and then ensure that we don’t comment on PRs that have any of the following:
+
+1. More than 7 files modified
+   - We cannot extract function information from deleted files because the Github API does not give us a patch, and it doesn’t make sense to count new files since they won’t have any issues associated with them
+2. More than 500 lines changed
+
+Why do we do this? At a certain point, when a PR is touching so many lines and/or files, it becomes less and less useful to point out specific issues related to the functions inside the PR. For instance, if somebody applies a linting change, it could result in a lot of lines modified.
+
+Note that we only support a particular set of languages (via looking at file extensions). We skip counting files and lines if they don’t have a file extension we support, and only continue with the files that we can support if these checks are met.
+
+### Fetching issues for each file
+
+#### Reverse codemapping
+
+1.  Normal codemappings map a file in the stack trace (within Sentry) to the source code (in a source code management integration such as Github) using a stack trace root and a source code root. Sentry may store filenames differently than in Github, so codemappings store this relationship. We leverage codemappings here because we want to figure out the Sentry project(s) associated with the file and the stored name of the file in order to make a Snuba query, which requires `project_id`.
```

Comment:
_codemappings map_ -> I work here and I'm not sure what this is, how about linking to the docs?